projects/packages/publicize/src/class-publicize.php

```diff
@@ -271,9 +290,57 @@ public function get_all_connections_for_user() {
 				}
 			}
 		}
+
+		if ( self::use_admin_ui_v1() && isset( $args['run_test_results'] ) && $args['run_test_results'] && count( $connections_to_return ) > 0 ) {
+			$connections_to_return = $this->add_connection_test_results( $connections_to_return );
+		}
+
 		return $connections_to_return;
 	}
 
+	/**
+	 * To add the connection test results to the connections.
+	 *
+	 * @param array $connections The Jetpack Social connections.
+
+	 * @return array
+	 */
+	public function add_connection_test_results( $connections ) {
+		$path                   = sprintf( '/sites/%d/publicize/connection-test-results', absint( Jetpack_Options::get_option( 'id' ) ) );
+		$response               = Client::wpcom_json_api_request_as_user( $path, '2', array(), null, 'wpcom' );
+		$connection_results     = json_decode( wp_remote_retrieve_body( $response ), true );
+		$connection_results_map = array();
+		foreach ( $connection_results as $connection_result ) {
+			$connection_results_map[ $connection_result['connection_id'] ] = $connection_result['test_success'];
+		}
+
+		return array_map(
+			function ( $connection ) use ( $connection_results_map ) {
+				if ( isset( $connection_results_map[ $connection['connection_id'] ] ) ) {
+						$connection['status'] = $connection_results_map[ $connection['connection_id'] ] ? 'ok' : 'broken';
+				}
+				return $connection;
+			},
+			$connections
+		);
```

Comment:
I love the array_ functions, but I wonder if it overcomplicates things here, and another `foreach` would be clearer and more efficient?

```suggestion
		foreach ( $connection_results as $connection_result ) {
			$connection_results_map[ $connection_result['connection_id'] ] = $connection_result['test_success'] ? 'ok' : 'broken';
		}

		foreach ( $connections as $connection ) {
			if ( isset( $connection_results_map[ $connection['connection_id'] ] ) ) {
				$connection['status'] = $connection_results_map[ $connection['connection_id'] ];
			}
		};
		
		return $connections;
```

Reply:
I could be wrong about this, but  I think `array_map` takes care of things deep copying. 

I originally wrote it like how you had proposed and tried it without your commit suggestion as well, and if the connection status is broken, it still retains the default value of ok, which is set on the connections array, which is passed as a param to this function,

If go the array map route, that problem disappears.

Reply:
That seems very odd! We are mutating the original array in this second one, which should work and be more memory efficient as we're not making a copy of it

Reply:
Oh we're not mutating the right copy! It needs to be something like:

```
		foreach ( $connections as $key => $connection ) {
			if ( isset( $connection_results_map[ $connection['connection_id'] ] ) ) {
				$connections[ $key ]['status'] = $connection_results_map[ $connection['connection_id'] ];
			}
		};
```

Reply:
Thanks, I jumped a gun there. It's fixed in https://github.com/Automattic/jetpack/pull/37510 now